src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -26,37 +26,93 @@
 #include <multipass/format.h>
 
 #include <iostream>
+#include <sstream>
 
 namespace mp = multipass;
 namespace mpl = multipass::logging;
 
 namespace
 {
 constexpr auto category = "sshfs mount";
-template <typename Callable>
-auto run_cmd(mp::SSHSession& session, std::string&& cmd, Callable&& error_handler)
+const std::string fuse_version_string{"FUSE library version "};
+const std::string ld_library_path_key{"LD_LIBRARY_PATH="};
+const std::string snap_path_key{"SNAP="};
+
+auto run_cmd(mp::SSHSession& session, std::string&& cmd)
 {
     auto ssh_process = session.exec(cmd);
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
+        throw std::runtime_error(ssh_process.read_std_error());
+
     return ssh_process.read_std_output();
 }
 
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
+auto match_line(std::istringstream& output, const std::string& matcher)
 {
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    output.seekg(0, output.beg);
+
+    std::string line;
+    while (std::getline(output, line, '\n'))
+    {
+        if (line.find(matcher) != std::string::npos)
+        {
+            return line;
+        }
+    }
+
+    return std::string{};
```

Comment:
MEH! ;P

Should this go into utils?

Reply:
Yeah, that's probably a better place for this.

Reply:
Yep, moved into utils now.  Will write a simple test for it.